tensorrt_llm/_torch/modules/attention.py

```diff
@@ -239,6 +242,47 @@ def forward(
         return attn_output
 
 
+def extract_extra_attrs(layer_idx: str):
+    extra_attrs = get_model_extra_attrs()
+    assert extra_attrs is not None, "Model extra attrs is not set"
+
+    metadata_ref = extra_attrs.get("attention_metadata", None)
+    assert metadata_ref is not None, "Attention metadata is not set"
+    metadata = metadata_ref()
+    assert isinstance(
+        metadata,
+        TrtllmAttentionMetadata,
+    )
+
+    mla_layers = extra_attrs.get("mla_layers", None)
+    assert mla_layers is not None, "MLA layers is not registered"
+    mla_layer_ref = mla_layers.get(layer_idx, None)
+    assert mla_layer_ref is not None, f"Cannot find MLA layer for layer {layer_idx}"
+    mla_layer = mla_layer_ref()
+    assert issubclass(type(mla_layer), MLA) or isinstance(
+        mla_layer,
+        MLA), "MLA layer must be a subclass of MLA or an instance of MLA"
```

Comment:
```suggestion
    assert isinstance(
        mla_layer,
        MLA), "MLA layer must be an instance of MLA"
```
See https://stackoverflow.com/questions/8107313/isinstance-and-issubclass-return-conflicting-results